lib/rules/rule-empty-line-before/index.js

```diff
@@ -153,6 +131,20 @@ const rule = function(expectation, options, context) {
   };
 };
 
+function isAfterRule(rule) {
+  const prevRule = rule.prev();
+  return prevRule && prevRule.type === "rule";
+}
+
+function isAfterSingleLineComment(rule) {
+  const prevRule = rule.prev();
```

Comment:
Should like use the `isSharedLineComment` util?

Reply:
Turns out there's already an `isAfterCommentLine` util so this function doesn't even need to exist.

Reply:
Whoops, never mind about that; `isAfterCommentLine` also counts multi-line comments.

Also, when I fix this function, a few tests start failing:

`rule-empty-line-before › accept › ["always", {"except": ["after-single-line-comment"]}]`

```css
@media { /* comment */
  a {}
}
```

`rule-empty-line-before › reject › ["always", {"except": ["after-single-line-comment"]}]`

```css
@media { /* comment */

  a {}
}
```

They actually need to be reversed now because the shared-line comment doesn't count as a single-line comment, so the first test should be rejected and the second one should be accepted. This might need to be highlighted in the release notes of the release that includes this fix.

Reply:
> They actually need to be reversed now because the shared-line comment doesn't count as a single-line comment

Sounds about right. I had assumed we weren't testing for this eventually. However, it turns out we are but the tests are wrong.

> This might need to be highlighted in the release notes of the release that includes this fix.

Good point. Due to our [Semantic Versioning Policy](https://github.com/stylelint/stylelint#semantic-versioning-policy) this _can be_ included in a minor release, even though it could break some users' builds, though.

Reply:
> Whoops, never mind about that; `isAfterCommentLine` also counts multi-line comments.

Feel free to create a `isAfterSingleLineComment` util, though.

And also free feel to rename the `isAfterCommentLine` util to `isAfterComment` for consistency with the other `isAfter*` utils. I'm not sure the `Line` suffix makes sense now that _shared-line_ comments are always ignored.